src/crimson/osd/osd_operation_sequencer.h

```diff
@@ -0,0 +1,102 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <map>
+
+#include "crimson/common/operation.h"
+#include "osd/osd_types.h"
+
+namespace crimson::osd {
+
+template <typename>
+struct OperationComparator;
+
+template <typename T>
+class OperationRepeatSequencer {
```

Comment:
i still think this class deserves some brief comments explaining what it is for.

1. we need to drain in-flight ops which are issued in previous PG interval before issuing new ones if PG advances to a new interval. (why)
2. ops targeting the same PG are ordered by their `id` . the id is guaranteed to be monotonically increasing and unique for each client request to a certain PG. we need to keep the same order in which ops are received. (why)

Reply:
when PG interval changes, we are supposed to interrupt all in-flight ops. but in the order in which the ops are interrupted are not determined because they are scheduled by the seastar scheduler, if we just interrupt them at seeing a different interval when moving to a new continuation. but we are supposed to replay the ops from the same client targeting the same PG in the exact order that they are received.

the way how we address this problem is to group the ops by PG and order them by the `id` which must be monotonically increasing and unique on PG basis.